src/NServiceBus.Core.Tests/DelayedDelivery/TimeoutManager/ExpiredTimeoutsPollerTests.cs

```diff
@@ -64,6 +67,30 @@ public void TearDown()
             Assert.AreEqual(currentTime + InMemoryTimeoutPersister.EmptyResultsNextTimeToRunQuerySpan, poller.NextRetrieval);
         }
 
+        [Test]
+        public async Task Polls_with_same_start_slice_when_dispatch_fails()
+        {
+            var failingDispatcher = new FailableDispatcher();
+            poller = new ExpiredTimeoutsPoller(timeouts, failingDispatcher, "test", breaker, () => currentTime);
+
+            RegisterNewTimeout(currentTime.Subtract(TimeSpan.FromMinutes(5)));
+
+            // fail first dispatch
+            failingDispatcher.DispatcherAction = _ => { throw new Exception("transport error"); };
+            Assert.ThrowsAsync<Exception>(() => poller.SpinOnce(CancellationToken.None));
```

Comment:
Since ` Assert.AreEqual(1, unicastTransportOperations.Count);` is the actual assert we care about I would naturally lean towards wrap it in a try / catch.